pkg/controller/applications/comp.go

```diff
@@ -18,5 +18,5 @@ func IsApplicationUpToDate(cr *v1alpha1.ApplicationParameters, remote *argocdv1a
 		// the unexported fields should not bother here, since we don't copy them or write them
 		cmpopts.IgnoreUnexported(argocdv1alpha1.ApplicationDestination{}),
 	}
-	return cmp.Equal(*cluster, remote.Spec, opts...) && cmp.Equal(cr.Annotations, remote.Annotations, opts...)
+	return cmp.Equal(*cluster, remote.Spec, opts...) && cmp.Equal(cr.Annotations, remote.Annotations, opts...) && cmp.Equal(cr.Finalizers, remote.Finalizers, opts...)
```

Comment:
afaik order of finalizers does not matter for K8s garbage collection. I think it makes sense to ignore ordering for this slice.

Instead of using `cmp` we can probably also use Go's build-in functionality like https://pkg.go.dev/slices#Equal and https://pkg.go.dev/slices#Sort.

(The same goes for `cr.Annotations` where we can use https://pkg.go.dev/maps#Equal)